pallets/nodle-staking/src/types.rs

```diff
@@ -0,0 +1,536 @@
+    }
+}
+
+#[derive(Encode, Decode, Clone, RuntimeDebug)]
+pub struct Nominator<AccountId, Balance> {
+    pub nominations: OrderedSet<Bond<AccountId, Balance>>,
+    pub total: Balance,
+    pub active_bond: Balance,
+    pub unlocking: Vec<UnlockChunk<Balance>>,
+}
+
+impl<
+        AccountId: Ord + Clone,
+        Balance: Copy
+            + AtLeast32BitUnsigned
+            + Saturating
+            + sp_std::ops::AddAssign
+            + sp_std::ops::Add<Output = Balance>
+            + sp_std::ops::SubAssign
+            + PartialOrd
+            + Default,
+    > Nominator<AccountId, Balance>
+{
+    pub fn new(validator: AccountId, amount: Balance) -> Self {
+        Nominator {
+            nominations: OrderedSet::from(vec![Bond {
+                owner: validator,
+                amount,
+            }]),
+            total: amount,
+            active_bond: amount,
+            unlocking: Vec::new(),
+        }
+    }
+    pub fn add_nomination(&mut self, bond: Bond<AccountId, Balance>) -> bool {
+        let amt = bond.amount;
+        if self.nominations.insert(bond) {
+            self.total = self.total.saturating_add(amt);
+            self.active_bond = self.active_bond.saturating_add(amt);
+            true
+        } else {
+            false
+        }
+    }
+    // Returns Some(remaining balance), must be more than MinNominatorStake
+    // Returns None if nomination not found
+    pub fn rm_nomination(&mut self, validator: AccountId) -> Option<Balance> {
+        let mut amt: Option<Balance> = None;
+        let nominations = self
+            .nominations
+            .0
+            .iter()
+            .filter_map(|x| {
+                if x.owner == validator {
+                    amt = Some(x.amount);
+                    None
+                } else {
+                    Some(x.clone())
+                }
+            })
+            .collect();
+        if let Some(balance) = amt {
+            self.nominations = OrderedSet::from(nominations);
+            self.active_bond = self.active_bond.saturating_sub(balance);
+            Some(self.active_bond)
+        } else {
+            None
+        }
+    }
+    // Returns None if nomination not found
+    pub fn inc_nomination(&mut self, validator: AccountId, more: Balance) -> Option<Balance> {
+        match self
+            .nominations
+            .0
+            .binary_search(&Bond::from_owner(validator.clone()))
+        {
+            Ok(loc) => {
+                let nom_bond = &mut self.nominations.0[loc];
+                nom_bond.amount = nom_bond.amount.saturating_add(more);
+                self.total = self.total.saturating_add(more);
+                self.active_bond = self.active_bond.saturating_add(more);
+                Some(nom_bond.amount)
+            }
+            Err(_) => None,
+        }
+    }
+    pub fn dec_nomination(
+        &mut self,
+        validator: AccountId,
+        less: Balance,
+    ) -> Option<Option<Balance>> {
```

Comment:
Is there no way we can avoid returning an `Option` wrapped into another `Option`?

Reply:
Indeed bit crap, right now signature of the api modified as below

```
    pub fn dec_nomination(
        &mut self,
        validator: AccountId,
        less: Balance,
    ) -> Result<Balance, String> {
```
Done :+1: